modules/apps/29-fee/keeper/grpc_query.go

```diff
@@ -1,5 +1,61 @@
 package keeper
 
-// TODO
+import (
+	"context"
 
-//var _ types.QueryServer = Keeper{}
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ types.QueryServer = Keeper{}
+
+// IncentivizedPackets implements the IncentivizedPackets gRPC method
+func (q Keeper) IncentivizedPackets(c context.Context, req *types.QueryIncentivizedPacketsRequest) (*types.QueryIncentivizedPacketsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	ctx = ctx.WithBlockHeight(int64(req.QueryHeight))
```

Comment:
Another small nit:
This could be reduced to a single line and might improve readability by chaining `.WithBlockHeight()` like so:

```go
ctx := sdk.UnwrapSDKContext(goCtx).WithBlockHeight(int64(req.QueryHeight))
```